src/components/Questionnaire/QuestionnaireEditor.tsx

```diff
@@ -105,6 +105,140 @@ import { SelectOrCreateValueset } from "./SelectOrCreateValueset";
 import ValueSetSelect from "./ValueSetSelect";
 import { scrollToQuestion } from "./utils";
 
+const MAX_FILE_SIZE = 10 * 1024 * 1024; // 10MB in bytes
+
+interface ValidationResult {
+  isValid: boolean;
+  error?: string;
+  data?: QuestionnaireDetail;
+}
+
+const validateFileSize = (size: number): boolean => {
+  return size <= MAX_FILE_SIZE;
+};
+
+const enableWhenSchema = z
+  .array(
+    z.object({
+      question: z.string().min(1, "Question reference is required"),
+      operator: z.string().min(1, "Operator is required"),
+      answer: z.any().optional(),
+    }),
+  )
+  .optional();
+
+const ImportquestionSchema: z.ZodType<any> = z.lazy(() =>
```

Comment:
Duplication w/ reg schema.

Instead create a base schema that can be used for both and then extend.